File: check_spot.py
import urllib.request
import json
import os

api_key = os.environ.get("VITE_EIA_KEY")
url = f"https://api.eia.gov/v2/petroleum/pri/spt/facet/series?api_key={api_key}"

req = urllib.request.Request(url)
with urllib.request.urlopen(req) as response:
    data = json.loads(response.read().decode())
    print(json.dumps(data["response"]["facets"], indent=2))

File: check_spot.js
import axios from 'axios';
import dotenv from 'dotenv';
dotenv.config();

const apiKey = process.env.VITE_EIA_KEY;

async function checkEia() {
    try {
        const url = `https://api.eia.gov/v2/petroleum/pri/spt/facet/series?api_key=${apiKey}`;
        const res = await axios.get(url);
        console.log(JSON.stringify(res.data.response.facets, null, 2));
    } catch (e) {
        console.error(e.message);
    }
}
checkEia();
